lib/ex_aws/ecs.ex

```diff
@@ -0,0 +1,1091 @@
+defmodule ExAws.ECS do
+
+  @namespace "AmazonEC2ContainerServiceV20141113"
+
+  @moduledoc """
+  Amazon EC2 Container Service (Amazon ECS) is a highly scalable, fast,
+  container management service that makes it easy to run, stop, and manage
+  Docker containers on a cluster of EC2 instances.
+
+  Amazon ECS lets you launch and stop container-enabled applications with simple API calls, allows you
+  to get the state of your cluster from a centralized service, and gives you
+  access to many familiar Amazon EC2 features like security groups, Amazon
+  EBS volumes, and IAM roles.
+
+  You can use Amazon ECS to schedule the placement of containers across your
+  cluster based on your resource needs, isolation policies, and availability
+  requirements. Amazon EC2 Container Service eliminates the need for you to
+  operate your own cluster management and configuration management systems or
+  worry about scaling your management infrastructure.
+  """
+
+  @doc """
+  Creates a new Amazon ECS cluster.
+
+  By default, your account receives a `default` cluster when you launch your first container instance. However,
+  you can create your own cluster with a unique name with the `create_cluster/1`
+  action.
+  """
+  @spec create_cluster(cluster_name :: binary) :: ExAws.Operation.JSON.t
+  def create_cluster(cluster_name) do
+    request(:create_cluster, %{"clusterName" => cluster_name})
+  end
+  @doc """
+  Creates a cluster with name `default`.
+
+  See also `create_cluster/1`.
+  """
+  @spec create_cluster() :: ExAws.Operation.JSON.t
+  def create_cluster() do
+    request(:create_cluster)
+  end
+
+  # http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PlacementConstraint.html
+  @type placement_constraint_type :: [
+    :distinctInstance |
+    :memberOf
+  ]
+  @type placement_constraint :: [
+    {:expression, binary} |
+    {:type, placement_constraint_type}
+  ]
+
+  # http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PlacementStrategy.html
+  @type placement_strategy_type :: [
+    :random |
+    :spread |
+    :binpack
+  ]
+  @type placement_strategy :: [
+    {:field, binary} |
+    {:type, placement_strategy_type }
+  ]
+
+  # http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LoadBalancer.html
+  @type load_balancer :: [
+    {:container_name, binary} |
+    {:container_port, 1..65535} |
+    {:load_balancer_name, binary} |
+    {:target_group_arn, binary}
+  ]
+
+  @type create_service_opts :: [
+    {:client_token, binary} |
+    {:cluster, binary} |
+    {:deployment_configuration, deployment_conf} |
+    {:load_balancers, list(load_balancer)} |
```

Comment:
Ok, no problem - I have no preference whatsoever, this is the first time I've used types.
I'll change it.